pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java

```diff
@@ -597,6 +596,39 @@ public static String splitPart(String input, String delimiter, int index) {
     }
   }
 
+  /**
+   * @param input the input String to be split into parts.
+   * @param delimiter the specified delimiter to split the input string.
+   * @param index the specified index for the splitted parts to be returned.
+   * @param max the max count of parts that the input string can be splitted into.
+   * @return splits string on the delimiter with the limit count and returns String at specified index from the split.
+   */
+  @ScalarFunction
+  public static String splitPart(String input, String delimiter, int index, int max) {
```

Comment:
I feel putting `max` in front of `index` is more intuitive

Reply:
gotcha, i was following the parameter orders of `StringUtils.splitByWholeSeparator(input, delimiter, max)`
i can update it.

Reply:
updated, thanks